sdk/eventhub/azure-eventhub/azure/eventhub/aio/_transport/_pyamqp_transport_async.py

```diff
@@ -262,7 +262,11 @@ async def _receive_task(consumer):
             while retried_times <= max_retries and running and consumer._callback_task_run:
                 try:
                     await consumer._open() # pylint: disable=protected-access
-                    running = await cast(ReceiveClientAsync, consumer._handler).do_work_async(batch=consumer._prefetch)
+                    if len(consumer._message_buffer) < max_batch_size:
+                        running = await cast(ReceiveClientAsync, consumer._handler).do_work_async(
+                            batch=consumer._prefetch
+                        )
+                    await asyncio.sleep(0.05)
```

Comment:
an asyncio.sleep(0) isn't enough to give the callback task time to grab the lock?

Reply:
the sleep is in effect only when there are no messages to send back. Otherwise we dont sleep at all, so getting the lock would be difficult as its a tight loop